db/attachment_test.go

```diff
@@ -1359,3 +1359,129 @@ func TestMigrateBodyAttachmentsMergeConflicting(t *testing.T) {
 	bodyAtts, foundBodyAtts := body1[BodyAttachments]
 	assert.False(t, foundBodyAtts, "not expecting '_attachments' in body but found them: %v", bodyAtts)
 }
+
+func TestAllowedAttachments(t *testing.T) {
+	defer base.SetUpTestLogging(base.LevelDebug, base.KeySync)()
+
+	var tests = []struct {
+		name              string
+		inputBlipProtocol string
+		inputAttVersion   int
+	}{
+		{"TestAllowedAttachmentsCBMobile2AttVer1", BlipCBMobileReplicationV2, AttVersion1},
+		{"TestAllowedAttachmentsCBMobile2AttVer2", BlipCBMobileReplicationV2, AttVersion2},
+		{"TestAllowedAttachmentsCBMobile3AttVer1", BlipCBMobileReplicationV3, AttVersion1},
+		{"TestAllowedAttachmentsCBMobile3AttVer2", BlipCBMobileReplicationV3, AttVersion2},
+	}
+
+	requireIsAttachmentAllowedTrue := func(t *testing.T, ctx *BlipSyncContext, docID string, meta []AttachmentStorageMeta, activeSubprotocol string) {
+		for _, att := range meta {
+			key := allowedAttachmentKey(docID, att.digest, activeSubprotocol)
+			require.True(t, ctx.isAttachmentAllowed(key))
+		}
+	}
+
+	requireIsAttachmentAllowedFalse := func(t *testing.T, ctx *BlipSyncContext, docID string, meta []AttachmentStorageMeta, activeSubprotocol string) {
+		for _, att := range meta {
+			key := allowedAttachmentKey(docID, att.digest, activeSubprotocol)
+			require.False(t, ctx.isAttachmentAllowed(key))
+		}
+	}
+
+	// Single document associated with multiple attachments of different digests.
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := &BlipSyncContext{}
+			meta := []AttachmentStorageMeta{
+				{digest: "digest1", version: tt.inputAttVersion},
+				{digest: "digest2", version: tt.inputAttVersion},
+			}
+			var docID = "doc1"
```

Comment:
When running with BlipCBMobileReplicationV2, we won't know the docID when calling isAttachmentAllowed.  Shouldn't the test be following that pattern as well?